x/merkledb/db.go

```diff
@@ -159,6 +159,21 @@ type MerkleDB interface {
 	Prefetcher
 }
 
+func NewConfig() Config {
```

Comment:
It feels a bit odd to default to populating `mockMetrics` - I'm wondering if the `newDefaultConfig` needs to exist at all or if we should just use this. (and populate the metrics reg here)..

Although that feels a bit odd as well... so 🤷 maybe the issue is with the `mockMetrics` as a whole...

It feels like whoever made `mockMetrics` didn't know that `"github.com/prometheus/client_golang/prometheus/testutil"` existed.